### DISCUSSION TITLE
How to reset faker.unique() on v6.3.1?

### DISCUSSION THREAD

**Discussion by aboutjquery** [faker-js/faker#1084] [Q&A]
I have a piece of code running on gulp that mostly creates lots of addresses for html to read and apply.

```
var address = [];

for (let i = 0; i < 50; i++) {
  address.push(faker.unique(faker.address.country));
}
```

My design is to execute the file and reload again when the file or the html itself has changed, I use gulp's watch method.
I get an error message when I change this file or html several times.

```
Error: Exceeded maxRetries: 10 for uniqueness check.
May not be able to generate any more unique values with current settings.
Try adjusting maxTime or maxRetries parameters for faker.unique().
```

I changed `i` to` i < 5` without pausing the watch but still got the error, until I stopped the watch and re-run the task, of course the error still appeared after several times.

I thought that `faker.unique()` would be reset every time I re-execute the above code, but obviously not, I believe there is something like cache behind it.

Refer to other articles 7.0 seems to have a solution but I can only use the 6.3.1 version, is there a solution?

Does `maxRetries` help with this?

Thanks.

**Comment by Shinigami92** [comment]
https://github.com/faker-js/faker/pull/800 was released in `v6.3.0` 🤔
So you should be able to pass your own store and with that you can fully control it.
It's really manual and has not a really nice DX, I know and sorry for that, but it solves the problem
You can manually clear all key-values in the store and then rerun the function

**Reply by aboutjquery** [reply]
Thats great because i am using v6.3.1, is there any simple example please? I'm really new to javascript skills and I don't know how to start.

I want to do reset store every time before executing the following code, so its wont show the maxRetries error.

```
const arr = [];

for (let i = 0; i < 50; i++) {
  const name = faker.unique(faker.name.firstName);
  arr.push(name);
}

console.log(arr);
```

 Thanks so much.

**Reply by ST-DDT** [reply]
You can always look at our documentation:
https://fakerjs.dev/api/unique.html

I'm not sure what you expect from the unique method if you don't want `maxRetries` errors.
If you reset the store every time you call the method, then you don't get unique results at all.
You could try to increase the retry counter though.

````js
const store = {};
faker.unique(faker.name.firstName, undefined, {
  // maxRetries: 50,
  // maxTime: 50,
  store: store,
})
````

Which locale are you using, that you encounter the limits that often?
